website/notifications/events/model.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+"""Basic Event handling for events that need subscriptions"""
+
+from furl import furl
+from datetime import datetime
+from six import add_metaclass
+
+from website.notifications import emails
+from website.notifications.constants import NOTIFICATION_TYPES
+from website.models import Node
+from website.notifications.events import utils as event_utils
+from website.notifications import utils
+
+
+class _EventMeta(type):
```

Comment:
Once you have that, you can also remove the `Event` code from `wiki.addons.base.views` and use a signal instead

```python
# website.addons.base.views
# NOTE: I gave little thought to naming the signal
file_action.send(user, node, action, payload, ...)
```

```python
# website.notifications.events.model

@file_action.connect
def send_file_updated_email(node, user, action, ...):

    EventClass  = event_map[action]
    event = EventClass()
    event.perform()
```